common/persistence/cassandra/mutable_state_task_store.go

```diff
@@ -153,6 +153,13 @@ const (
 		`and task_id >= ? ` +
 		`and task_id < ?`
 
+	templateIsQueueEmptyQuery = `SELECT task_id ` +
+		`FROM executions ` +
+		`WHERE shard_id = ? ` +
+		`and type = ? ` +
+		`and namespace_id = ? ` +
+		`limit 1`
```

Comment:
I hope we won't have lots of dlq tasks and encounter tombstone issues...

Reply:
I need to investigate in another approach. tombstone is a valid concern.

Reply:
Added range filter